components/renku_data_services/session/api.spec.yaml

```diff
@@ -539,6 +589,38 @@ components:
       description: A description for the resource
       type: string
       maxLength: 500
+    BuildParameters:
+      description: Build parameters
+      type: object
+      additionalProperties: false
+      properties:
+        repository:
+          $ref: "#/components/schemas/Repository"
+        builder_variant:
+          $ref: "#/components/schemas/BuilderVariant"
+        frontend_variant:
+          $ref: "#/components/schemas/FrontendVariant"
+      required:
+        - repository
+        - builder_variant
+        - frontend_variant
+    BuildParametersPost:
```

Comment:
We need to keep `BuildParametersPatch` as well to allow partial update of the parameters.

Reply:
It forces ALL parameters to be given, no partials.

Reply:
Right, I've added a `BuildParametersPatch` with optional fields.